src/handlers/events/delivery/goals/forkGoal.ts

```diff
@@ -0,0 +1,199 @@
+/*
+ * Copyright © 2018 Atomist, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+    AutomationContextAware,
+    HandlerContext,
+    HandlerResult,
+    logger,
+    Secrets,
+} from "@atomist/automation-client";
+import { automationClientInstance } from "@atomist/automation-client/automationClient";
+import { ApolloGraphClient } from "@atomist/automation-client/graph/ApolloGraphClient";
+import {
+    EventIncoming,
+    RequestProcessor,
+} from "@atomist/automation-client/internal/transport/RequestProcessor";
+import { RegistrationConfirmation } from "@atomist/automation-client/internal/transport/websocket/WebSocketRequestProcessor";
+import { guid } from "@atomist/automation-client/internal/util/string";
+import { AutomationEventListenerSupport } from "@atomist/automation-client/server/AutomationEventListener";
+import { QueryNoCacheOptions } from "@atomist/automation-client/spi/graph/GraphClient";
+import * as cluster from "cluster";
+import * as fs from "fs-extra";
+import {
+    OnAnyRequestedSdmGoal,
+    ProgressLog,
+    SdmGoalById,
+} from "../../../..";
+import { SdmGoalImplementationMapper } from "../../../../common/delivery/goals/SdmGoalImplementationMapper";
+import { ProjectLoader } from "../../../../common/repo/ProjectLoader";
+import { StringCapturingProgressLog } from "../../../../spi/log/ProgressLog";
+import { spawnAndWatch } from "../../../../util/misc/spawned";
+import { FulfillGoalOnRequested } from "./FulfillGoalOnRequested";
+
+export type IsolatedGoalLauncher = (goal: OnAnyRequestedSdmGoal.SdmGoal,
```

Comment:
The file name doesn't seem to reflect the content. Also, if I understand correctly this file mixes a specific implementation (K8s) with the generic concept. Can we split the K8s implementation out?

Reply:
Done